weldx/geometry.py

```diff
@@ -276,11 +276,10 @@ def rasterize(self, raster_width: pint.Quantity) -> pint.Quantity:
             Array of raster points
 
         """
+        raster_width = Q_(raster_width)
         if raster_width <= 0:
             raise ValueError("'raster_width' must be a number greater than 0.")
-        num_pts = (
-            np.round((self._length / raster_width).to_base_units()).astype(int) + 1
-        )
+        num_pts = np.round((self._length / raster_width).m).astype(int) + 1
```

Comment:
This doesn't work reliably when units differ, see the following examples:
```python
import numpy as np
from weldx import Q_

_length = Q_("10m")
raster_width = Q_("2mm")

num_pts = np.round((_length / raster_width).magnitude).astype(int) + 1
num_pts = max(num_pts, 2)
num_pts
#> 6

num_pts = np.round((_length / raster_width)).astype(int) + 1
num_pts = max(num_pts, 2)
num_pts
#> <Quantity(5001.0, 'dimensionless')>

num_pts = np.round(_length / raster_width).to_base_units().magnitude.astype(int) + 1
num_pts = max(num_pts, 2)
num_pts
#> 5001

vals = np.linspace(0.0, 1.0, num=num_pts, endpoint=True)
vals
#> array([0.000e+00, 2.000e-04, 4.000e-04, ..., 9.996e-01, 9.998e-01,
#>        1.000e+00])
```

Using `.magnitude` without `.to()` or `.to_base_units()` first is too unreliable

Reply:
yes that is what happens